arch/x86/adler32_avx.c

```diff
@@ -33,30 +49,18 @@ Z_INTERNAL uint32_t adler32_avx2(uint32_t adler, const unsigned char *buf, size_
     if (UNLIKELY(len < 16))
         return adler32_len_16(adler, buf, len, sum2);
 
-    uint32_t ALIGNED_(32) s1[8], s2[8];
-
-    memset(s1, 0, sizeof(s1)); s1[7] = adler; // TODO: would a masked load be faster?
-    memset(s2, 0, sizeof(s2)); s2[7] = sum2;
+    const __m256i vs_mask = _mm256_setr_epi32(0, 0, 0, 0, 0, 0, 0, -1);
+    __m256i vs1 = _mm256_set1_epi32(adler);
+    __m256i vs2 = _mm256_set1_epi32(sum2);
+    vs1 = _mm256_and_si256(vs1, vs_mask);
+    vs2 = _mm256_and_si256(vs2, vs_mask);
```

Comment:
I'm not 100% sure if it would be better or not to group setting up `vs1` first and then setting up `vs2` after that, so it matches how it is done below.

All four lines are short and the readability wouldn't hurt much from reducing the line count from 4 to 2.

Reply:
As far as the line count goes I think the way it is is more readable.

Reply:
I'm just thinking aloud... Currently it's just inconsistent with the block below in regards to grouping the lines... 

But also, in here we have to declare the types of the variables, which isn't necessary below.

Reply:
I tend to like to group declarations together.  I'm willing to reorder the block below it, since that's probably what the instruction scheduling stage of the optimizer should be doing anyway, since both of these operations can be performed on different execution ports in the same clock cycle (though, the OoO nature of the architecture is probably doing this speculatively, anyway).

Reply:
Good now?

Reply:
Yup... I don't want to be too picky, I just made one final review now that I have no projects that need my attention...